### PR TITLE
Reading view mode: Basic functionality

### LINKED ISSUE #45: As a user, I can view a saved link in a clean reading view
so I can focus on the content with no distractions.

#### Part of epic: 
#27 Reader view

### PR DESCRIPTION
**Pull request by rlrh**
#### How to use:
`view reader` for reading view
`view browser` for browser view
#### How it works:
When reading view is selected, `BrowserPanel` loads the original web page first, then inputs the resulting HTML into Crux for processing, and finally loads the cleaned-up HTML.

### CODE REVIEW

**Comment by rlrh** [comment]
Test coverage dropped by 0.7% as it is very difficult to test for certain cases.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -334,13 +334,18 @@ public boolean equals(Object obj) {
...
                 && Objects.equals(selectedPerson.get(), other.selectedPerson.get())
                 && Objects.equals(currentViewMode.get(), other.currentViewMode.get())
-                && Objects.equals(commandResult.get(), other.commandResult.get())
-                && exception.get().getMessage().equals(other.exception.get().getMessage());
+                && Objects.equals(commandResult.get(), other.commandResult.get());
+        if (exception.get() == null && other.exception.get() == null) {
+            return stateCheck;
+        } else {
+            return stateCheck && exception.get().getMessage().equals(other.exception.get().getMessage());
+        }
```
Comment (thomastanck) [inline src/main/java/seedu/address/model/ModelManager.java:353]: It's possible one exception is null but the other isn't.

File: src/main/java/seedu/address/commons/util/XmlUtil.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        TransformerFactory tf = TransformerFactory.newInstance();
+        Transformer transformer = tf.newTransformer();
+        transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+        StringWriter writer = new StringWriter();
+        transformer.transform(new DOMSource(doc), new StreamResult(writer));
+        String output = writer.getBuffer().toString();
+        return output;
+    }
+
+    private static Document convertStringToDocument(String xmlStr) throws Exception {
```
Comment (thomastanck) [inline src/main/java/seedu/address/commons/util/XmlUtil.java]: `throws Exception` do we have a more specific type here?

File: src/main/java/seedu/address/ui/BrowserPanel.java [hunk 3]
```diff
@@ -42,70 +43,117 @@
     private WebEngine webEngine = browser.getEngine();
 
     private String currentLocation;
-    private boolean isCurrentlyNonBrowserView;
+    private boolean isCurrentlyReaderView;
     private ViewMode viewMode;
+    private Consumer<CommandResult> onSuccess;
```
Comment (thomastanck) [inline src/main/java/seedu/address/ui/BrowserPanel.java]: We'll merge this for now but note this is why we usually don't want to do any processing on the UI side :p

Comment (rlrh) [reply src/main/java/seedu/address/ui/BrowserPanel.java]: The most recent commit doesn't have the CommandResult consumer any more.

File: src/main/java/seedu/address/model/ModelManager.java (hunk 1 above)
Comment (rlrh) [reply src/main/java/seedu/address/model/ModelManager.java:353]: Oh yeah, I'll fix this.

File: src/main/java/seedu/address/commons/util/XmlUtil.java (hunk 2 above)
Comment (rlrh) [reply src/main/java/seedu/address/commons/util/XmlUtil.java]: There's a couple: `javax.xml.parsers.ParserConfigurationException`, `org.xml.sax.SAXException`, `java io.IOException`

**Review by thomastanck — APPROVED** [review]
lgtm

**Pull request merged**